feature-libs/tms/gtm/services/gtm.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable, isDevMode, OnDestroy } from '@angular/core';
+import { CxEvent, EventService, WindowRef } from '@spartacus/core';
+import { TmsConfig } from '@spartacus/tms/core';
+import { merge, Subscription } from 'rxjs';
+
+export type GtmDataLayer = any[];
+
+export interface GtmWindow extends Window {
+  dataLayer?: GtmDataLayer;
+}
+
+/**
+ * This service interacts with the configured data layer object by pushing the Spartacus events to it.
+ */
+@Injectable({ providedIn: 'root' })
+export class GoogleTagManagerService implements OnDestroy {
+  protected subscription = new Subscription();
+
+  constructor(
+    protected eventsService: EventService,
+    protected windowRef: WindowRef,
+    protected tmsConfig: TmsConfig
+  ) {}
+
+  /**
+   * This method is called only once to start collecting and dispatching events to GTM
+   */
+  collect(): void {
+    this.prepareDataLayer();
```

Comment:
I guess we should not run it on the sever side

Reply:
It's not running, as it checks for the window inside of it.

Reply:
Checking for the window is not a good check for SSR (as we realized in the past)

Reply:
You're right, done.